### PR TITLE
GCI 1459 add description details to mid api

### PR DESCRIPTION
**Pull request by rjadhav-ch**
This adds description details to a MID item.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/logging/LoggingUtils.java [hunk 1]
```diff
@@ -81,4 +82,16 @@ public static void logErrorWithStatus(final Map<String, Object> logMap,
...
+
+    /**
+     * Method to log an error and add a value for the description key.
+     *
+     * @param descriptionKey the company number to log under the key {@link LoggingUtils#DESCRIPTION_LOG_KEY}
+     * @param errorMessage the error message to display
+     */
+    public static void logOrdersDescriptionsConfigError(final String descriptionKey, final String errorMessage) {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(DESCRIPTION_LOG_KEY, descriptionKey);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/logging/LoggingUtils.java:94]: This `logMap` is unused?

File: src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderService.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    /**
+     * Gets the configured description.
+     * @param companyNumber the company number making up part of the description
+     * @return the configured description, or <code>null</code> if none found.
+     */
+    public String getDescription(final String companyNumber) {
+        if (missingImageDeliveryDescription == null) {
+            // Error logged again here at time description is requested.
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderService.java:48]: What does this comment refer to?

File: src/test/postman/Missing_Image_Delivery_API.postman_collection.json [hunk 3]
```diff
@@ -151,7 +151,7 @@
 					"_postman_isSubFolder": true
 				},
 				{
-					"name": "Create certified copy item",
+					"name": "Create Missing Image Delivery item",
```
Comment (VaughanJackson123) [inline src/test/postman/Missing_Image_Delivery_API.postman_collection.json:154]: 👍

File: src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderService.java (hunk 2 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderService.java]: removed. thanks

File: src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/logging/LoggingUtils.java (hunk 1 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/missingimagedelivery/orders/api/logging/LoggingUtils.java:94]: updated. nice catch. thanks

File: src/test/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * Unit tests the {@link DescriptionProviderService} class.
+ */
+class DescriptionProviderServiceTest {
+
+    private static final String COMPANY_NUMBER = "00006400";
+    private static final String EXPECTED_DESCRIPTION = "missing image delivery for company " + COMPANY_NUMBER;
+
+    @Test
+    @DisplayName("Provides expected description")
+    void getDescriptionProvidesExpectedDescription() {
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderServiceTest.java:22]: This test seems to fail for me.

Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/missingimagedelivery/orders/api/service/DescriptionProviderServiceTest.java:22]: You will need to update api-enumerations submodule.

**Review by VaughanJackson123 — APPROVED** [review]

**Pull request merged**